process_qsos_dr7.m

```diff
@@ -0,0 +1,476 @@
+prior.z_qsos  = all_zqso_dr7(prior_ind);
+prior.c4_ind = prior_ind;
+prior.z_c4 = all_z_civ_c13(prior_ind);
+
+% filter out CIVs from prior catalog corresponding to region of spectrum below
+% Ly-alpha QSO rest. In Roman's code, for detecting DLAs, instead of
+% Ly-alpha they have Lyman-limit
+for i = size(prior.z_c4)
+    if (observed_wavelengths(civ_1548_wavelength , prior.z_c4(i)) < ...
+            observed_wavelengths(min_lambda, prior.z_qsos(i)))
+        prior.c4_ind(i) = false;
+    end
+end
+prior = rmfield(prior, 'z_c4');
+
+% enable processing specific QSOs via setting to_test_ind
+if (ischar(test_ind))
+    test_ind = eval(test_ind);
+end
+
+all_wavelengths    =    all_wavelengths(test_ind);
+all_flux           =           all_flux(test_ind);
+all_noise_variance = all_noise_variance(test_ind);
+all_pixel_mask     =     all_pixel_mask(test_ind);
+all_sigma_pixel    =    all_sigma_pixel(test_ind);
+z_qsos             =           all_zqso_dr7(test_ind);
+num_quasars        =                numel(z_qsos);
+
+% preprocess model interpolants
+% griddedInterpolant does an interpolation and gives a function handle
+% based on the grided data. If the data is 2xD, {x,y} are the the same size as
+% row  and columns. M is like M=f(x,y) and is like a matrix with each element
+% M(i,j) = f(x(i), y(j))
+mu_interpolator = ...
+    griddedInterpolant(rest_wavelengths,        mu,        'linear');
+M_interpolator = ...
+    griddedInterpolant({rest_wavelengths, 1:k}, M,         'linear');
+
+% initialize results with nan
+min_z_c4s                   = nan(num_quasars, 1);
+max_z_c4s                   = nan(num_quasars, 1);
+log_priors_no_c4            = nan(num_quasars, 1);
+log_priors_c4               = nan(num_quasars, 1);
+log_likelihoods_no_c4       = nan(num_quasars, 1);
+sample_log_likelihoods_c4L1 = nan(num_quasars, num_C4_samples);
+sample_log_likelihoods_c4L2 = nan(num_quasars, num_C4_samples);
+log_likelihoods_c4L1        = nan(num_quasars, 1);
+log_likelihoods_c4L2        = nan(num_quasars, 1);
+log_posteriors_no_c4        = nan(num_quasars, 1);
+log_posteriors_c4L1         = nan(num_quasars, 1);
+log_posteriors_c4L2         = nan(num_quasars, 1);
+map_N_c4L1                  = nan(num_quasars, 1);
+map_N_c4L2                  = nan(num_quasars, 1);
+map_z_c4L1                  = nan(num_quasars, 1);
+map_z_c4L2                  = nan(num_quasars, 1);
+map_sigma_c4L1              = nan(num_quasars, 1);
+map_sigma_c4L2              = nan(num_quasars, 1);
+EW1_fine                    = nan(num_quasars, 1);
+EW2_fine                    = nan(num_quasars, 1);
+EW1_spline                  = nan(num_quasars, 1);
+EW2_spline                  = nan(num_quasars, 1);
+data_civ                    = nan(num_quasars, 1);
+fitted_continuum            = cell(num_quasars,1);
+
+if RejectionSampling==0
+    sample_sigma_c4 = min_sigma + (max_sigma-min_sigma)*offset_sigma_samples;
+else
+    sample_sigma_c4 = sigma_samples;
+end
+fAVG= 1/(2*nAVG + 1);
```

Comment:
Could you explain this line?

Reply:
This is a number that you use it when you calculate the averaging. I defined it outside of loop for avoiding division multiple times in a loop. I think it is not that much effective in increasing the performance.

Reply:
So this number is how much you would like to increase the resolution after applying `finer`. I guess this is how much the pixel width you want to decrease?